src/common/interface.ts

```diff
@@ -8,4 +8,13 @@ export default interface IRoute {
 
 export interface IParams {
   id?: string | undefined;
-}
\ No newline at end of file
+}
+
+export interface CustomStyleProps {
+  width?: number;
+  height?: number;
+  flexDirection?: string;
+  placeholder?: string;
+  justifyContent?: string;
+  alignItems?: string;
+};
```

Comment:
Looking at the current usage of this type, don't think it generalises well. Let's define individual types containing only the props we need in the respective component files.

Reply:
It is one interface for all uses. 
If I define individual types then there will be a lot of overlaps.
This way it seems more flexible, as it will enable to change of multiple style properties with just a single interface.

It sort of enables more flexibility for styled.components.
Any new CSS property change via props will need a change in the interface.

So using a common style interface.
The most regularly used CSS properties are in there.

Let me know if this makes sense.

Reply:
Correct. It's exactly because of that flexibility I want to avoid a common type. We want to add that friction so that when someone add a new CSS property to the specific interface Typescript will warn us about it.

Reply:
Sure. I will make the changes.